src/huggingface_hub/hf_api.py

```diff
@@ -7607,6 +7611,14 @@ def create_inference_endpoint(
                 Secret values to inject in the container environment.
             type ([`InferenceEndpointType]`, *optional*):
                 The type of the Inference Endpoint, which can be `"protected"` (default), `"public"` or `"private"`.
+            domain (`str`, *optional*):
+                The custom domain for the Inference Endpoint (e.g. `"my-new-domain.cool-website.woof"`).
+            path (`str`, *optional*):
+                The custom path for the Inference Endpoint, should start with a `/` (e.g. `"/models/google-bert/bert-base-uncased"`).
```

Comment:
What does it means exactly? Is it that the endpoint should be served under `endpoints-(...)-url.co/models/google-bert/bert-base-uncased` or is it that the URL `endpoints-(...)-url.co/` points to the `/models/google-bert/bert-base-uncased` path of the docker image? (worth some clarification in the docstring IMO)

Reply:
IIRC @oOraph mentioned this is applicable for anyone with a custom domain registered as well, so I'd consider this to be generic.

Reply:
Yes, worth having the parameter supported in `huggingface_hub`! Just wanted to be sure what it meant exactly (_The custom path for the Inference Endpoint_ is not explicit enough IMO)

Reply:
updating now

Reply:
updated here: https://github.com/huggingface/huggingface_hub/pull/3013/commits/54572d5dce4bf3a2061cd83706b7b444e54938ad

Reply:
i think it's when you want to expose the resulting endpoint's inference route under a custom path but you can check with @oOraph and @XciD